sufia-models/app/services/sufia/characterization_service.rb

```diff
@@ -0,0 +1,67 @@
+module Sufia
+  class CharacterizationService
+    include Hydra::Derivatives::ExtractMetadata
+
+    delegate :mime_type, :uri, to: :@generic_file
+    attr_reader :generic_file
+
+    def self.run(generic_file)
+      new(generic_file).characterize
+    end
+
+    def initialize(generic_file)
+      @generic_file = generic_file
+    end
+
+    ## Extract the metadata from the content datastream and record it in the characterization datastream
+    def characterize
+      store_metadata(extract_metadata)
+      store_fulltext(extract_fulltext)
+      generic_file.filename = [generic_file.content.original_name]
+      generic_file.save
+    end
+
+    protected
+
+      def store_fulltext(extracted_text)
+        generic_file.full_text.content = extracted_text if extracted_text.present?
+      end
+
+      def extract_fulltext
+        FullTextExtractionService.run(@generic_file)
```

Comment:
Should this be stored in a class variable so that it only gets run once.  I can see you are only running it once now, but in the future the method could get called multiple times.  Some one wanting to add logging ect.